server: de-flake TestHealthAPI

tbg opened the pull request:
Temporary errors can be returned when intentionally horking up the node.

Fixes #18469.

(or exposes a permanent failure of this test masked by the timeout).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19213)
<!-- Reviewable:end -->

tbg committed:
server: de-flake TestHealthAPI

Temporary errors can be returned when intentionally horking up the node.

Fixes #18469.

(or exposes a permanent failure of this test masked by the timeout).

a-robinson reviewed (approved):
Seems worth a shot. LGTM
